File: 033 OOPS in C++/004 Constructor 3.cpp
#include <bits/stdc++.h>
class Student {
  
    public:
    
    string name;
    int roll_no;
    int marks;
    
    Student() {
        cout<<"Student Class Constructor called\n";
    }
    
    Student(string name,int roll_no, int marks) {
        this->name = name;
        this->roll_no = roll_no;
        this-> marks = marks;
    }
    
    Student(Student &other) {
        this->name = other.name;
        this->roll_no = other.roll_no;
        this->marks = other.marks;
    }
    
    void setData(string name,int roll_no, int marks) {
        this->name = name;
        this->roll_no = roll_no;
        this-> marks = marks;
    }
    
    void printData() {
        cout<<"My name is "<<name<<" and my roll number is "<<roll_no<<". I got "<<marks<<" marks out of 100\n";
    }
    
};

int main() {
    
    //parameterized constructor is being called
    Student s1("Guneet",2,95);
    s1.printData();
    
  //copy constructor is being called
    Student s2(s1);
    s2.printData();
}

File: 032 OOPS in Java/004 Constructor 3.java
import java.util.*;

class Student {
    String name;
    int marks;
    int rollNo;
    
    Student() {
        System.out.println("An object of Student class is created.");
    }
    
    Student(String name, int marks, int rollNo) {
        this.name = name;
        this.marks = marks;
        this.rollNo = rollNo;
    }
    
    Student(Student other) {
        this.name = other.name;
        this.marks = other.marks;
    }
    
    void displayData() {
        System.out.println("I am " + name + " and my rollNo is " + rollNo + ". I got " + marks + " marks out of 100.");
    }
}

public class Main {
    public static void main(String[] args) {
        Student s1 = new Student("Guneet",95,2);
        s1.displayData();
        Student s2 = new Student(s1);
        s2.rollNo = 1;
        s2.displayData();
    }    
}
